src/sentry/integrations/slack/tasks/find_channel_id_for_rule.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from collections.abc import Sequence
+from typing import Any
+
+from sentry.incidents.models.alert_rule import AlertRuleTriggerAction
+from sentry.integrations.services.integration import integration_service
+from sentry.integrations.slack.utils import (
+    SLACK_RATE_LIMITED_MESSAGE,
+    RedisRuleStatus,
+    strip_channel_name,
+)
+from sentry.integrations.slack.utils.channel import get_channel_id_with_timeout
+from sentry.mediators.project_rules.creator import Creator
+from sentry.mediators.project_rules.updater import Updater
+from sentry.models.project import Project
+from sentry.models.rule import Rule, RuleActivity, RuleActivityType
+from sentry.shared_integrations.exceptions import ApiRateLimitedError, DuplicateDisplayNameError
+from sentry.silo.base import SiloMode
+from sentry.tasks.base import instrumented_task
+
+logger = logging.getLogger("sentry.integrations.slack.tasks")
+
+
+@instrumented_task(
+    name="sentry.integrations.slack.search_channel_id",
```

Comment:
I'm seeing two tasks with this name. One in this file and the other in `src/sentry/tasks/integrations/slack/find_channel_id_for_rule`. We should have unique names for celery tasks.